djangosaml2idp/views.py

```diff
@@ -34,31 +37,60 @@
 except AttributeError:
     raise ImproperlyConfigured("SAML_IDP_SPCONFIG not defined in settings.")
 
+user_identity = getattr(settings, 'SAML_IDP_IDENTIFYING_ATTRIBUTE', 'username')
+
+
+def build_url(*args, **kwargs):
+    get = kwargs.pop('get', {})
+    url = reverse(*args, **kwargs)
+    if get:
+        url += '?' + urllib.parse.urlencode(get)
+    return url
+
 
 @never_cache
 @csrf_exempt
-@require_http_methods(["GET", "POST"])
-def sso_entry(request):
+@require_http_methods(["POST"])
+def sso_post(request):
+    data = request.POST
+    entry_binding = BINDING_HTTP_POST
+    return sso_entry(request, data, entry_binding)
+
+
+@never_cache
+@require_http_methods(["GET"])
+def sso_redirect(request):
+    data = request.GET
+    entry_binding = BINDING_HTTP_REDIRECT
+    return sso_entry(request, data, entry_binding)
+
+
+def sso_entry(request, data, entry_binding):
     """ Entrypoint view for SSO. Gathers the parameters from the HTTP request, stores them in the session
         and redirects the requester to the login_process view.
     """
-    passed_data = request.POST if request.method == 'POST' else request.GET
+    passed_data = data
     try:
         request.session['SAMLRequest'] = passed_data['SAMLRequest']
     except (KeyError, MultiValueDictKeyError) as e:
         return HttpResponseBadRequest(e)
-    request.session['RelayState'] = passed_data.get('RelayState', '')
     # TODO check how the redirect saml way works. Taken from example idp in pysaml2.
     if "SigAlg" in passed_data and "Signature" in passed_data:
         request.session['SigAlg'] = passed_data['SigAlg']
         request.session['Signature'] = passed_data['Signature']
-    return HttpResponseRedirect(reverse('djangosaml2idp:saml_login_process'))
+    relayState = passed_data.get('RelayState', '')
+    if relayState == 'None':
+        relayState = ""
+    url = build_url('djangosaml2idp:saml_login_process',
+                    get={'binding': entry_binding, 'RelayState': relayState})
+    return HttpResponseRedirect(url)
 
 
 class IdPHandlerViewMixin:
     """ Contains some methods used by multiple views """
 
-    error_view = import_string(getattr(settings, 'SAML_IDP_ERROR_VIEW_CLASS', 'djangosaml2idp.error_views.SamlIDPErrorView'))
+    error_view = import_string(getattr(
```

Comment:
I'm not really a fan of most of these formatting changes. Would you mind to revert these and only keep the changes for the functionality? I don't think splitting lines in these changes improves readability; I prefer single lines (I don't follow the 80-char belief). If it is too much of a hassle (maybe because your favorite code editor does it automatically?), feel free to say so, I can change them myself after merging.